app/frontend/stores/permit-application-store.ts

```diff
@@ -43,7 +43,8 @@ export const PermitApplicationStoreModel = types
   .actions((self) => ({
     __beforeMergeUpdate(permitApplicationData) {
       const pad = permitApplicationData
-      if (!pad?.jurisdiction?.id) return pad
+      // return unaltered data if any extended fields are not present, indicating base view
+      if (!pad?.jurisdiction?.id || !pad.stepCode?.id || !pad.submitter?.id) return pad
```

Comment:
Just clarifying if all permit applications are expected to have a stepCode? Because if the jurisdiction and submitter is present but stepCode is missing this will still early return